Stacking table component

gestchild opened the pull request:
References #6971

Adds a new responsive table component (styled as[ designs](https://app.zeplin.io/project/5aab926b4b6efde01259e959/screen/612de6c97e13d916c5085075)) which stacks cell rows on small screens. 

Uses this component on the account page, so we can fix the following issues in the above ticket and have it look good at all screen sizes.

- Item requests table should handle long titles appropriately (max width on column + text wrapping)
- show item title below work title

Before (small screens):
![small-b4](https://user-images.githubusercontent.com/6051896/134533026-ff933abd-cc08-49e9-a721-9b3260b3cd60.png)

After (small screens):
![small-after](https://user-images.githubusercontent.com/6051896/134533090-a20ff5fa-3c8a-4f42-a8a4-8863c04a3def.png)

Before (large screens):
![desk-b4](https://user-images.githubusercontent.com/6051896/134533172-3ec1ae53-be4d-46d9-b38f-9b6a753ad9ff.png)

After (large screens):
![Screenshot 2021-09-23 at 15 18 41](https://user-images.githubusercontent.com/6051896/134533214-a7c37c4f-8322-4024-8845-684d8e871647.png)

DominiqueMarshall commented:
I'm happy with this, the only thing missing from the designs is this additional text change to the cancelling holds sentence:
**_"Requests made will be available to pick up from the library for two weeks. If you wish to cancel a hold, please contact the library team."_**

gestchild commented:
> I'm happy with this, the only thing missing from the designs is this additional text change to the cancelling holds sentence:
> **_"Requests made will be available to pick up from the library for two weeks. If you wish to cancel a hold, please contact the library team."_**

I've updated the text and changed the font, as discussed

davidpmccormick reviewed (approved):
🚀
